src/main/java/org/terasology/taskSystem/DelayedNotificationSystem.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.taskSystem;
+
+import org.terasology.engine.Time;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.chat.ChatMessageEvent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.registry.In;
+import org.terasology.registry.Share;
+
+@Share(DelayedNotificationSystem.class)
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class DelayedNotificationSystem extends BaseComponentSystem {
+    @In
+    private Time time;
+
+    @In
+    private LocalPlayer localPlayer;
+
+    /**
+     * The maximum delay time after which notification is sent again.
+     */
+    private long maxDelay = 10;
+
+    public float sendNotification(String message, EntityRef notificationMessageEntity, float lastNotification) {
+        if (time.getGameTime() - lastNotification > maxDelay || lastNotification == 0) {
```

Comment:
You should check for lastNotification == 0 first, then the subtraction > max check.